javascript/exercises/creditcard.html

```diff
@@ -1,58 +1,101 @@
+                <li>The final digit must be even</li>
+                <li>The sum of all the digits must be greater than 16</li>
+            </ul>
+            </p>
+
+            <p>The following credit card numbers are valid:</p>
+            <ul>
+                <li><code>9999777788880000</code></li>
+                <li><code>6666666666661666</code></li>
+            </ul>
+
+            <p>The following credit card numbers are invalid:
+            <ul>
+                <li><code>a92332119c011112</code> <em>invalid characters</em></li>
+                <li><code>4444444444444444</code> <em>only one type of number</em></li>
+                <li><code>1111111111111110</code> <em>sum less than 16</em></li>
+                <li><code>6666666666666661</code> <em>odd final number</em></li>
+            </ul>
+            </p>
+        </section>
+        <section>
+            <p>In order to run the function, you'll need to know how to <a
+                    href="loadJavascriptTemplate.html" target="_blank">load javascript on an HTML page.</a> From
+                there, you will open your developer console to call the function.</p>
+            <img src="../images/console.png" style="max-width: 400px"/>
+        </section>
+        <hr/>
+        <section>
+            <p>
+                <strong>Bonus #1:</strong> A valid credit card number may also contain dashes, to make a card number
+                easier to read. For example, the following credit card numbers are now also valid:
+            <ul>
+                <li><code>9999-7777-8888-0000</code></li>
+                <li><code>6666-6666-6666-1666</code></li>
+            </ul>
+            Update your program to allow such numbers.
+            (Hint: Remove the dashes from the input string before checking if the input credit card number is
+            valid.)
+            </p>
+
+            <p>
+                <strong>Bonus #2:</strong> Return an object indicating whether the credit card is valid, and if not,
+                what
+                the error
+                is
+                <br>
+                <code>{ valid: true, number: 'a923-3211-9c01-1112' }</code>
+                <br>
+                <code>{ valid: false, number: 'a923-3211-9c01-1112', error: ‘wrong_length’ }</code>
+            </p>
+
+            <p><strong>Bonus #3:</strong> Make your credit card scheme even more advanced! What are the rules, and what
+                are
+                some numbers that pass or fail? Ideas: check expiration date! Check out the <a
+                        href="https://en.wikipedia.org/wiki/Luhn_algorithm" target="_blank">Luhn Algorithm</a> for
+                inspiration.
+            </p>
+        </section>
+        <section>
+            <button class="btn" onclick="showSolution(1)">See Solution</button>
+        </section>
+    </main>
```

Comment:
@meganesu You'll need to add the link to the solution as div with id #solution1 for the main exercise solution #solution2 for the bonus #2, etc with style display: none. See the showSolution() function for what it expects.